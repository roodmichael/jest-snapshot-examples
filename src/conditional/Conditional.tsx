import * as React from 'react';

export interface ConditionalProps { isLoggedIn: boolean};

const LoggedIn = () => <h1>Oh, Hai.</h1>;
const NotLoggedIn = () => <h1>Stranger danger.</h1>;

export const Conditional = (props: ConditionalProps) => {
    if (props.isLoggedIn) {
        return <LoggedIn />
    }
    return <NotLoggedIn />;
};
