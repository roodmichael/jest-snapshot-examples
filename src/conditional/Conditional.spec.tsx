import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Conditional } from './Conditional';

it('shows hello message when logged in.', () => {
  const tree = renderer.create(
    <Conditional isLoggedIn />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('shows stranger danger message when not logged in.', () => {
  const tree = renderer.create(
    <Conditional isLoggedIn={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
