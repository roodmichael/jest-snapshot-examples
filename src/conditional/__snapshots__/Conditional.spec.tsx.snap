// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows hello message when logged in. 1`] = `
<h1>
  Oh, Hai.
</h1>
`;

exports[`shows stranger danger message when not logged in. 1`] = `
<h1>
  Stranger danger.
</h1>
`;
